app/controllers/tag_controller.rb

```diff
@@ -525,6 +525,17 @@ def comments
     render 'show'
   end
 
+  def subtopics
+    # here, we look for this special tag marking a node as having a parent
+    # collect the "slugs" i.e. the unique part of the URL, which should correspond to a tagname
+    subtopic_names = Tag.find_by(name: "parent:#{params[:topic]}")
+      &.nodes
+      &.collect(&:slug)
```

Comment:
OK, I pushed a couple commits to fix minor issues. See how I moved this up out of the "private" section.

Also I added `&.` which fails gracefully if the value is nil, instead of erroring - it just doesn't continue trying. 

I think it should work now, but I think it's worth writing a functional test for this. That will mean adding a tag to a wiki page with name `parent:TAGNAME` such that the code actually gets used. 

The reason is that for most tags we might test this with, there won't be any subtopics. We need to create some test data to make that situation appear so that we can write a test around it. Make sense? I'm happy to help with the test data, let me know if you'd like some support on that!

Reply:
And great work getting this far. It's complex!!

Reply:
@jywarren 
I tried opening the page: `/tags/subtopic/one`, I am able to successfully render the page (blank although), and I can render topic name also on the page by extracting it from the URL. I created a wiki page with tag `parent:one` but I am still unable to fetch subtopics on the page. I doubt if `@subtopics` actually carries the subtopics in it. 

Also, a mention, on line 531 here, we have  

```ruby
Tag.find_by(name: "parent:#{params[:topic]}")
```

Here, tag is compared with `parent:#{params[:topic]}`, right? So I also tried creating a wiki page with tag `parent:#one` but still can't fetch it on the `/tags/subtopic/one` page. Conversely, I also tried to change the line 531 to this - 

```ruby
Tag.find_by(name: "parent:{params[:topic]}")
```

and created a wiki page with tag `parent:one`, but still I can't obtain desired result.

Reply:
So, do you think I have this test data (the wiki pages I created), so as to write a test? Also, I am totally new to writing tests, could you please guide me with it? PS: Sharing resources also would do.

Thanks!

Reply:
Hi @KarishmaVanwari ok so you are doing really well then, if it rendered a blank page. We just need to debug the controller. 

A controller test is known as a functional test -- there is documentation here, but it's pretty dense: https://guides.rubyonrails.org/testing.html#functional-tests-for-your-controllers

I might start by looking at other tests:

https://github.com/publiclab/plots2/blob/4204a62b5bfff77f0e2bb3ce3bcb68a4b3a6d984/test/functional/tag_controller_test.rb#L206-L214

And at how we create test data -- by creating entries in https://github.com/publiclab/plots2/blob/main/test/fixtures/

For tags, we'll have to create not only a tag fixture entry in https://github.com/publiclab/plots2/blob/main/test/fixtures/tags.yml  

but also a record to link that tag to a node: https://github.com/publiclab/plots2/blob/main/test/fixtures/node_tags.yml

Does that make sense? Let's start with that, can you add that to this PR? Once it's in place, I can help you write a simple test, then narrow the test assertions until we find what was wrong with our assumptions. How does that sound?

Reply:
And, as far as I can tell, your `parent:one` tag should work! And here is hopefully a helpful tip - the `"line of text with #{something} in the middle"` syntax is a way to insert Ruby code into a string. It must use " double quotes, and the inserted Ruby code to be evaluated is surrounded by `#{  }`. It's a little weird looking but it is smoother than adding a bunch of separate strings together.

Reply:
@jywarren : I created test data in https://github.com/publiclab/plots2/blob/main/test/fixtures/tags.yml and https://github.com/publiclab/plots2/blob/main/test/fixtures/node_tags.yml. Let me know if I got that right. Thanks!